client/state/selectors/is-passwordless-jetpack-connection-flow.ts

```diff
@@ -0,0 +1,28 @@
+import { get } from 'lodash';
+import getCurrentQueryArguments from 'calypso/state/selectors/get-current-query-arguments';
+import getInitialQueryArguments from 'calypso/state/selectors/get-initial-query-arguments';
+import { isWooCommerceCoreProfilerFlow } from './is-woocommerce-core-profiler-flow';
+import type { AppState } from 'calypso/types';
+
+const isWooCommercePaymentsOnboardingFlow = ( state: AppState ): boolean => {
+	return (
+		( get( getInitialQueryArguments( state ), 'from' ) === 'woocommerce-payments' ||
+			get( getCurrentQueryArguments( state ), 'from' ) === 'woocommerce-payments' ) &&
+		( get( getInitialQueryArguments( state ), 'plugin_name' ) === 'woocommerce-payments' ||
+			get( getCurrentQueryArguments( state ), 'plugin_name' ) === 'woocommerce-payments' )
```

Comment:
Nit: I think we should declare `get( getInitialQueryArguments( state ), 'from' )` and `get( getInitialQueryArguments( state ), 'plugin_name' )` as variables here, I feel it's too long to read clearly

Reply:
@ilyasfoo Good call! Updated it in [Clean up isWooCommercePaymentsOnboardingFlow function](https://github.com/Automattic/wp-calypso/pull/95504/commits/aaa955b3b1ed5cfcd010ce8f092311530ab01b39)